R/map-if-at.R

```diff
@@ -116,28 +125,20 @@ map_depth_rec <- function(.x,
                           .atomic) {
   if (.depth < 0) {
     abort("Invalid depth")
-  }
-
-  if (.atomic) {
-    if (!.ragged) {
-      abort("List not deep enough")
+  } else if (.depth == 0) {
+    .f(.x, ...)
+  } else if (.depth == 1) {
+    map(.x, .f, ...)
+  } else {
+    if (vec_is_list(.x)) {
```

Comment:
Probably we still want to recurse into data frames, since `map()` treats them as lists?

Reply:
I do want `vec_is_list()`, but purrr _may_ require special-casing data frames.

Reply:
I think maybe we should stick with `is.list()` for now; it feels like we need to ensure that `pluck()`, `pluck_depth()`, and `map_depth()` should all be consistent. 

Or maybe we should have `purrr_is_list()` which would be `vec_is_list(x) || is.data.frame(x)`? But that doesn't work for `pluck()` since it makes sense to (e.g.) pluck elements out of a linear model.

Reply:
yeah, that's probably safest. Ideally you'd either use accessors or unclass before plucking, with the understanding that you're reaching into unexported internals, but I guess it's too late to be this strict.

hmm... But did `is_atomic()` use to allow reaching into linear models with `map_depth()`?

Reply:
I think so, just because of the way we flipped testing for a leaf vs. testing for recursion.